modules/common-fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/common/fs2/io/Clients.scala

```diff
@@ -69,6 +70,24 @@ object Clients {
       .withRequestTimeout(readTimeout)
       .withMaxTotalConnections(maxConnections)
       .resource
+      .map(Retry[F](retryPolicy, redactHeadersWhen))
+
+  private def retryPolicy[F[_]] =
+    RetryPolicy[F](
+      backoff,
+      retriable = {
+        case (_, Left(Command.EOF)) => true
+        case _ => false
+      }
+    )
+
+  //retry immediately, max 10 times
+  private def backoff(attemptNumber: Int): Option[FiniteDuration] =
+    if (attemptNumber > 10) None
```

Comment:
I paused for a long time on this number "10".  I wonder if it should instead be "1".

Because the specific error case we know about is fixed by a tcp re-connect.  If it is not fixed by the first re-connect, then that's a _new_ error case that we don't know about.  And I wondered if it's better to let such a case raise an error, so at least we discover it.

But it's a tough call, and probably could be argued either way.  I'm happy to approve this as-is.